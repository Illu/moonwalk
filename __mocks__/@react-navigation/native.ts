import { Theme, DefaultTheme as defaultTheme } from "@react-navigation/native";

export const LightTheme = {
  ...defaultTheme,
  colors: {
    ...defaultTheme.colors,
    secondary: "#000",
    background: "#f1f2f7",
    text: "#000",
    placeholderText: "#7d7f86",
    inputBackground: "#dee1e7",
    secondaryText: "#666",
    accent: "#0a84ff",
    accentBackground: "rgba(10, 132, 255, 0.1)",
    uiAccent: "#c3c4c6",
  },
};

export const DefaultTheme = LightTheme;

export const DarkTheme: Theme = LightTheme;

export const useTheme = jest.fn().mockReturnValue({
  LightTheme,
});

export const useFocusEffect = jest.fn();
export const useNavigation = jest.fn().mockReturnValue({
  navigate: jest.fn(),
  replace: jest.fn(),
  reset: jest.fn(),
  setOptions: jest.fn(),
});
export const useNavigationState = jest.fn();
export const useRoute = jest.fn().mockReturnValue({
  params: {},
});
