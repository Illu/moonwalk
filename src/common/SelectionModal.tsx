import { useTheme } from "@react-navigation/native";
import React, { useEffect } from "react";
import { Modal, Animated } from "react-native";
import styled from "styled-components/native";

import { Browsers } from "../types";
import Icon from "./Icon";

const Wrapper = styled.SafeAreaView`
  background: rgba(0, 0, 0, 0.5);
  flex: 1;
  padding: 8px;
  justify-content: flex-end;
`;

const ButtonsWrapper = styled.View`
  justify-content: center;
  align-items: center;
  min-height: 50px;
  border-radius: 10px;
  margin: 5px 10px;
  background: ${({ theme }) => theme.colors.secondary};
`;

const Button = styled.TouchableOpacity<{ border: boolean }>`
  padding: 15px 20px;
  width: 100%;
  align-items: center;
  flex-direction: row;
  border-top-width: ${({ border }) => (border ? 1 : 0)}px;
  border-color: ${({ theme }) => theme.colors.uiAccent};
`;

const ButtonTitle = styled.Text`
  margin-left: 20px;
  flex: 1;
  font-size: 16px;
  color: ${({ theme }) => theme.colors.accent};
`;

const CancelTitle = styled.Text`
  font-weight: bold;
  font-size: 15px;
  align-self: center;
  color: ${({ theme }) => theme.colors.accent};
`;

const ModalTitle = styled.Text`
  padding: 15px 0;
  color: ${({ theme }) => theme.colors.secondaryText};
`;

interface Action {
  title: string;
  icon: string;
  action: () => any;
  id: Browsers;
}

interface Props {
  closeModal: () => any;
  actions: Action[];
  selected: Browsers;
  title: string;
}

const SelectionModal: React.FC<Props> = ({
  closeModal,
  title,
  actions,
  selected,
}) => {
  const { colors } = useTheme();

  const appearAnim = new Animated.Value(300);

  useEffect(() => {
    Animated.timing(appearAnim, {
      toValue: 0,
      duration: 200,
    }).start();
  });

  return (
    <Modal presentationStyle="overFullScreen" animationType="fade" transparent>
      <Wrapper>
        <Animated.View
          style={{
            transform: [
              {
                translateY: appearAnim,
              },
            ],
          }}
        >
          <ButtonsWrapper>
            <ModalTitle>{title}</ModalTitle>
            {actions.map((action) => (
              <Button
                key={action.id}
                border
                onPress={() => {
                  action.action();
                  closeModal();
                }}
              >
                <Icon name={action.icon} color={colors.accent} />
                <ButtonTitle>{action.title}</ButtonTitle>
                {selected === action.id && (
                  <Icon name="CheckCircle" color={colors.accent} />
                )}
              </Button>
            ))}
          </ButtonsWrapper>
          <ButtonsWrapper>
            <Button onPress={closeModal} style={{ justifyContent: "center" }}>
              <CancelTitle>Cancel</CancelTitle>
            </Button>
          </ButtonsWrapper>
        </Animated.View>
      </Wrapper>
    </Modal>
  );
};

export default SelectionModal;
