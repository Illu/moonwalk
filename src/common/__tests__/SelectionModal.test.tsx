import * as navigation from "@react-navigation/native";
import React from "react";

import TestRenderer from "../../helpers/testRenderer";
import { lightTheme } from "../../theme";
import { Browsers } from "../../types";
import SelectionModal from "../SelectionModal";

jest.mock("../../common/Icon", () => "Icon");
jest.useFakeTimers();

jest.spyOn(navigation, "useTheme");
navigation.useTheme.mockImplementation(() => lightTheme);

describe("SelectionModal", () => {
  it("renders correctly", () => {
    const props = {
      closeModal: jest.fn(),
      title: "$_TITLE_TEXT_$",
      actions: [
        {
          title: "$_TITLE_TEXT_$",
          icon: "$_ICON_NAME_$",
          action: jest.fn(),
          id: Browsers.safari,
        },
      ],
      selected: Browsers.safari,
    };
    const tree = TestRenderer(<SelectionModal {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
