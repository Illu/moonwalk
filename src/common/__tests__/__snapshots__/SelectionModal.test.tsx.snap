// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionModal renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  presentationStyle="overFullScreen"
  transparent={true}
  visible={true}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "translateY": 300,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 5,
              "minHeight": 50,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#666",
                "paddingBottom": 15,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 15,
              },
            ]
          }
        >
          $_TITLE_TEXT_$
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#c3c4c6",
              "borderTopWidth": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingBottom": 15,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 15,
              "width": "100%",
            }
          }
        >
          <Icon
            color="#0a84ff"
            name="$_ICON_NAME_$"
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#0a84ff",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "marginLeft": 20,
                },
              ]
            }
          >
            $_TITLE_TEXT_$
          </Text>
          <Icon
            color="#0a84ff"
            name="CheckCircle"
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 5,
              "minHeight": 50,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#c3c4c6",
              "borderTopWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 15,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 15,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#0a84ff",
                  "fontSize": 15,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</Modal>
`;
